### PR TITLE
Improve Mender state logic

### PR DESCRIPTION
**Pull request by pasinskim**
Add new update verification and rollback states.
Improve logging and fix default log level.
Refactor log and status sending. Fix some infinite loops while storing deployment logs.
Fix sending empty logs to server.
Update tests and documentation.

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacf @GregorioDiStefano please take a look. There is quite a lot of changes, but I have verified manually that everything is working. 
I will add missing tests soon.

File: state.go [hunk 1]
```diff
@@ -585,33 +583,29 @@ func (a *AuthorizedState) Handle(ctx *StateContext, c Controller) (State, bool)
 
 		// update prosess was initialized but stopped in the middle
 	case MenderStateUpdateFetch:
+		fallthrough
 	case MenderStateUpdateInstall:
```
Comment (kacf) [inline state.go]: You can use multiple values in one case block, like this:

```
case MenderStateUpdateFetch, MenderStateUpdateInstall:
```

File: state.go [hunk 2]
```diff
@@ -307,7 +307,7 @@ func (uv *UpdateVerifyState) Handle(ctx *StateContext, c Controller) (State, boo
 		return NewUpdateErrorState(NewTransientError(err), uv.update), false
 	}
 
-	log.Debugf("handle update verify state")
+	log.Info("handle update verify state")
```
Comment (kacf) [inline state.go]: Did you mean to include this? It looks quite debug-like to me.

**Comment by kacf** [comment]
It's hard to review all of it in detail, but looks good on the surface.

File: state.go (hunk 1 above)
Comment (pasinskim) [inline state.go]: Yup. Will change that.

File: state.go (hunk 2 above)
Comment (pasinskim) [inline state.go]: This is a small work-around for now if we are having no logs at all. I have added other logs to be stored in file, so can change the log level here.

**Comment by pasinskim** [comment]
Verified with end to end tests. Merging now!

**Pull request merged**